### ISSUE TITLE
[stable/nginx-ingress] [emerg] 46#46: bind() to 0.0.0.0:80 failed (13: Permission denied)

### ISSUE THREAD

**Issue by spursy** [helm/charts#15994]
- my kubernetes version is V14
- my node system is ubuntu 16

when i install stable/nginx-ingress, I encounter the followed issue:

```
ingress git:(master) kubectl logs pods/nginx-ingress-controller-b6754b88b-8vs24
-------------------------------------------------------------------------------
NGINX Ingress controller
  Release:    0.25.0
  Build:      git-1387f7b7e
  Repository: https://github.com/kubernetes/ingress-nginx
-------------------------------------------------------------------------------

I0731 11:48:09.768237       5 flags.go:192] Watching for Ingress class: nginx
W0731 11:48:09.768532       5 flags.go:221] SSL certificate chain completion is disabled (--enable-ssl-chain-completion=false)
nginx version: openresty/1.15.8.1
W0731 11:48:09.778373       5 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0731 11:48:09.778641       5 main.go:183] Creating API client for https://10.32.0.1:443
I0731 11:48:09.790363       5 main.go:227] Running in Kubernetes cluster version v1.14 (v1.14.0) - git (clean) commit 641856db18352033a0d96dbc99153fa3b27298e5 - platform linux/amd64
I0731 11:48:09.798778       5 main.go:91] Validated default/nginx-ingress-default-backend as the default backend.
I0731 11:48:09.918339       5 main.go:102] Created fake certificate with PemFileName: /etc/ingress-controller/ssl/default-fake-certificate.pem
E0731 11:48:09.921074       5 main.go:131] v1.14.0
W0731 11:48:09.938808       5 store.go:616] Unexpected error reading configuration configmap: configmaps "nginx-ingress-controller" not found
I0731 11:48:09.955841       5 nginx.go:275] Starting NGINX Ingress controller
I0731 11:48:11.157261       5 nginx.go:319] Starting NGINX process
I0731 11:48:11.158977       5 leaderelection.go:235] attempting to acquire leader lease  default/ingress-controller-leader-nginx...
W0731 11:48:11.161061       5 controller.go:384] Service "default/nginx-ingress-default-backend" does not have any active Endpoint
I0731 11:48:11.161568       5 controller.go:133] Configuration changes detected, backend reload required.
I0731 11:48:11.166200       5 status.go:86] new leader elected: nginx-ingress-controller-9f9fd9855-bfjfl
E0731 11:48:11.238432       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg884794920 syntax is ok
2019/07/31 11:48:11 [emerg] 32#32: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg884794920 test failed

-------------------------------------------------------------------------------
W0731 11:48:11.238764       5 queue.go:130] requeuing initial-sync, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg884794920 syntax is ok
2019/07/31 11:48:11 [emerg] 32#32: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg884794920 test failed

-------------------------------------------------------------------------------
I0731 11:48:14.494420       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:14.537840       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg351982951 syntax is ok
2019/07/31 11:48:14 [emerg] 39#39: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg351982951 test failed

-------------------------------------------------------------------------------
W0731 11:48:14.538240       5 queue.go:130] requeuing kube-system/kube-scheduler, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg351982951 syntax is ok
2019/07/31 11:48:14 [emerg] 39#39: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg351982951 test failed

-------------------------------------------------------------------------------
I0731 11:48:17.827731       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:17.878283       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg356635546 syntax is ok
2019/07/31 11:48:17 [emerg] 46#46: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg356635546 test failed

-------------------------------------------------------------------------------
W0731 11:48:17.878580       5 queue.go:130] requeuing kube-system/kube-controller-manager, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg356635546 syntax is ok
2019/07/31 11:48:17 [emerg] 46#46: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg356635546 test failed

-------------------------------------------------------------------------------
E0731 11:48:20.992760       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:21.161150       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:21.204013       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg891293233 syntax is ok
2019/07/31 11:48:21 [emerg] 53#53: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg891293233 test failed

-------------------------------------------------------------------------------
W0731 11:48:21.204316       5 queue.go:130] requeuing kube-system/tiller-deploy, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg891293233 syntax is ok
2019/07/31 11:48:21 [emerg] 53#53: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg891293233 test failed

-------------------------------------------------------------------------------
I0731 11:48:24.494524       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:24.546179       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg394277340 syntax is ok
2019/07/31 11:48:24 [emerg] 60#60: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg394277340 test failed

-------------------------------------------------------------------------------
W0731 11:48:24.546502       5 queue.go:130] requeuing default/kubernetes, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg394277340 syntax is ok
2019/07/31 11:48:24 [emerg] 60#60: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg394277340 test failed

-------------------------------------------------------------------------------
E0731 11:48:26.444714       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:27.827689       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:27.871995       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg615537035 syntax is ok
2019/07/31 11:48:27 [emerg] 67#67: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg615537035 test failed

-------------------------------------------------------------------------------
W0731 11:48:27.872287       5 queue.go:130] requeuing kube-system/kube-dns, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg615537035 syntax is ok
2019/07/31 11:48:27 [emerg] 67#67: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg615537035 test failed

-------------------------------------------------------------------------------
E0731 11:48:30.992809       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:31.161085       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:31.203381       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg298225518 syntax is ok
2019/07/31 11:48:31 [emerg] 74#74: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg298225518 test failed

-------------------------------------------------------------------------------
W0731 11:48:31.203693       5 queue.go:130] requeuing default/nginx-ingress-controller, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg298225518 syntax is ok
2019/07/31 11:48:31 [emerg] 74#74: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg298225518 test failed

-------------------------------------------------------------------------------
I0731 11:48:34.494327       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:34.536389       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg488150005 syntax is ok
2019/07/31 11:48:34 [emerg] 81#81: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg488150005 test failed

-------------------------------------------------------------------------------
W0731 11:48:34.536683       5 queue.go:130] requeuing default/nginx-ingress-default-backend, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg488150005 syntax is ok
2019/07/31 11:48:34 [emerg] 81#81: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg488150005 test failed

-------------------------------------------------------------------------------
E0731 11:48:36.444722       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:37.827733       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:37.870574       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg103017680 syntax is ok
2019/07/31 11:48:37 [emerg] 88#88: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg103017680 test failed

-------------------------------------------------------------------------------
W0731 11:48:37.870869       5 queue.go:130] requeuing default/nginx-ingress-controller-b6754b88b-8vs24, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg103017680 syntax is ok
2019/07/31 11:48:37 [emerg] 88#88: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg103017680 test failed

-------------------------------------------------------------------------------
E0731 11:48:40.992918       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:41.161141       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:41.204363       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg991448559 syntax is ok
2019/07/31 11:48:41 [emerg] 95#95: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg991448559 test failed

-------------------------------------------------------------------------------
W0731 11:48:41.204630       5 queue.go:130] requeuing default/nginx-ingress-controller-b6754b88b-8vs24, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg991448559 syntax is ok
2019/07/31 11:48:41 [emerg] 95#95: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg991448559 test failed

-------------------------------------------------------------------------------
I0731 11:48:44.494497       5 controller.go:133] Configuration changes detected, backend reload required.
E0731 11:48:44.544154       5 controller.go:145] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg656727170 syntax is ok
2019/07/31 11:48:44 [emerg] 102#102: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg656727170 test failed

-------------------------------------------------------------------------------
W0731 11:48:44.544474       5 queue.go:130] requeuing initial-sync, err
-------------------------------------------------------------------------------
Error: exit status 1
nginx: the configuration file /tmp/nginx-cfg656727170 syntax is ok
2019/07/31 11:48:44 [emerg] 102#102: bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)
nginx: configuration file /tmp/nginx-cfg656727170 test failed

-------------------------------------------------------------------------------
E0731 11:48:46.444631       5 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I0731 11:48:46.462385       5 main.go:154] Received SIGTERM, shutting down
I0731 11:48:46.462450       5 nginx.go:402] Shutting down controller queues
I0731 11:48:46.462470       5 status.go:117] updating status of Ingress rules (remove)
I0731 11:48:46.512434       5 status.go:136] removing address from ingress status ([])
I0731 11:48:46.512748       5 nginx.go:418] Stopping NGINX process
2019/07/31 11:48:46 [notice] 103#103: signal process started
I0731 11:48:47.539634       5 nginx.go:431] NGINX process has stopped
I0731 11:48:47.539873       5 main.go:162] Handled quit, awaiting Pod deletion
```

**Comment by fbreckle** [comment]
did you change controller.image.allowPrivilegeEscalation from true to false?

**Comment by spursy** [comment]
@fbreckle  I use default allowPrivilegeEscalation value

**Comment by fbreckle** [comment]
can you try putting
```securityContext:
          capabilities:
            add:
            - NET_BIND_SERVICE
            drop:
            - ALL
          procMount: Default
          runAsUser: 33```
into spec.template.spec.containers[0], (i.e. between resources: and terminationMessagePath)

**Comment by spursy** [comment]
@fbreckle  After added, I still encounter the same issue.

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
  creationTimestamp: "2019-08-08T08:11:35Z"
  generation: 1
  labels:
    app: nginx-ingress
    chart: nginx-ingress-1.12.1
    component: controller
    heritage: Tiller
    release: nginx-ingress
  name: nginx-ingress-controller
  namespace: default
  resourceVersion: "2997838"
  selfLink: /apis/extensions/v1beta1/namespaces/default/deployments/nginx-ingress-controller
  uid: 2395d0fd-b9b4-11e9-8dca-42010af0000a
spec:
  progressDeadlineSeconds: 2147483647
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: nginx-ingress
      component: controller
      release: nginx-ingress
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nginx-ingress
        component: controller
        release: nginx-ingress
    spec:
      containers:
      - args:
        - /nginx-ingress-controller
        - --default-backend-service=default/nginx-ingress-default-backend
        - --election-id=ingress-controller-leader
        - --ingress-class=nginx
        - --configmap=default/nginx-ingress-controller
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        image: quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.25.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz
            port: 10254
            scheme: HTTP
          initialDelaySeconds: 10
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        name: nginx-ingress-controller
        ports:
        - containerPort: 80
          hostPort: 80
          name: http
          protocol: TCP
        - containerPort: 443
          hostPort: 443
          name: https
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz
            port: 10254
            scheme: HTTP
          initialDelaySeconds: 10
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        resources: {}
        securityContext:
          allowPrivilegeEscalation: true
          capabilities:
            add:
            - NET_BIND_SERVICE
            drop:
            - ALL
          procMount: Default
          runAsUser: 33
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      hostNetwork: true
      nodeSelector:
        kubernetes.io/hostname: worker-0
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: nginx-ingress
      serviceAccountName: nginx-ingress
      terminationGracePeriodSeconds: 60
      tolerations:
      - operator: Exists
status:
  conditions:
  - lastTransitionTime: "2019-08-08T08:11:35Z"
    lastUpdateTime: "2019-08-08T08:11:35Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  observedGeneration: 1
  replicas: 1
  unavailableReplicas: 1
  updatedReplicas: 1
```

**Comment by aegershman** [comment]
Seeing the exact same thing after upgrading my cluster's VM types. Zero configuration changes in the chart or cluster, simply did a rolling upgrade of workers/masters to a different instance type. Not trying to hijack, just saying I'm seeing the exact same thing with almost exact same configuration OP has.

**Comment by fbreckle** [comment]
I had that problem when I set allowPrivilegeEscalation to false. Nginx-ingress has to run as root if it wants to bind to ports 80 and 443.
From the logs, for you guys, it cannot do that, even with allowPrivilegeEscalation set to true.
Do you have something in place in your cluster that prevents stuff from running with elevated permissions. PodSecurityPolicies maybe?

**Comment by wu0407** [comment]
```
 securityContext:
          allowPrivilegeEscalation: true
          capabilities:
            add:
            - NET_BIND_SERVICE
            drop:
            - ALL
          procMount: Default
          runAsUser: 33
```
try this has same problem

**Comment by wu0407** [comment]
https://github.com/containerd/containerd/issues/2516

**Comment by aegershman** [comment]
I found a solution to the problem. It might not be the same problem as you, but alas--

tl;dr, [the commands in the `Dockerfile` on `RUN setcap ...`](https://github.com/kubernetes/ingress-nginx/blob/master/rootfs/Dockerfile#L50-L54) give the `nginx-ingress` docker image certain permissions to talk to the root OS. To solve it, I had to SSH onto the k8s host and do `docker stop <image> && delete <image>`, etc., to purge the built image so the `RUN setcap` layer would run again... I wonder if setting `pullPolicy: Always` would help? Idk.

Some more background on the way I solved this--

The docker daemon running on each k8s host caches images in _layers_. So if the layer which represents `RUN setcap...` has already been ran and cached, then the final constructed image for `kubernetes-ingress` _will not run that layer again_, thus not perform `RUN setcap` again.

In my case, I started getting these errors when doing an in-place upgrade of the workers in my k8s cluster. I'm not entirely sure why, but I believe that during the upgrade, the permissions configured via `RUN setcap` were not present on the new set of workers. And the `RUN setcap` never ran again because it was cached as a `layer` in the docker image, and thus was never ran.

**Comment by ayoubensalem** [comment]
Hi, Is there any news ?

**Comment by aegershman** [comment]
I solved it by scaling down the number of ingress controllers to `0` (or uninstalling the chart entirely), ssh'ing onto each worker, and performing `docker rmi <ingress image>`.  I realized my problem was because the nginx docker image does some stuff during startup to grant itself permission on the OS. But those operations are performed as a docker layer; if those layers are cached, they won't run again. This problem happened to me when I changed the instance size of the workers to something larger; BOSH does this by removing the VM volume and remounting it to the new VM. So the cached docker image was there, but the operations the docker image _performed_ did not run again: https://github.com/kubernetes/ingress-nginx/blob/master/rootfs/Dockerfile#L46-L50 TL;DR, had to remove the image so that the image would be rebuilt.

Even if that solves your problem it's still a problem, but maybe try that? ... Or maybe try setting "imagePullPolicy: Always" and see if that forces the image to be rebuilt?

**Comment by liogate** [comment]
Hello, this command helped me:
`docker rm -f <container_id_of_ingress_having_troubles> && docker rmi rancher/nginx-ingress-controller:nginx-0.25.1-rancher1`

**Comment by dgmrdr** [comment]
Hi,

It can be caused by docker don't have rights to open pod's http/https ports as non-root user.
Modifing unprivilleged ports on node ( VM ) is **USELESS**, because problem is probably on side of POD network.

try to add : 
 hostNetwork: true

into
deployments/daemon-set/nginx-ingress.yaml,

or 
deployments/deployment/nginx-ingress.yaml,

```
    17     spec:
    18       serviceAccountName: nginx-ingress
    19       hostNetwork: true
    20       containers:
    21       - image: nginx/nginx-ingress:1.6.0
    22         name: nginx-ingress
    23         ports:
    24         - name: http
    25           containerPort: 80
    26           hostPort: 80
    27         - name: https
    28           containerPort: 443
    29           hostPort: 443
```

Then try to run deployment / daemon set.
This resolved my problem, the nginx-ingress started and the error disappeared from the logs

**Comment by martinfar** [comment]
I **solved** the issue running the following commands on each worker nodes 
`sudo fuser -k 80/tcp
sudo fuser -k 443/tcp
`
seems an old port rule bind was stuck

**Comment by zwpaper** [comment]
I run into this problem after changing the docker `root-dir`.

fixed following @aegershman guide:
1. change the controller replicaset to 0
2. docker rmi <image>
3. change the controller replicaset to the origin value

**Comment by mvits** [comment]
> I solved it by scaling down the number of ingress controllers to `0` (or uninstalling the chart entirely), ssh'ing onto each worker, and performing `docker rmi <ingress image>`. I realized my problem was because the nginx docker image does some stuff during startup to grant itself permission on the OS. But those operations are performed as a docker layer; if those layers are cached, they won't run again. This problem happened to me when I changed the instance size of the workers to something larger; BOSH does this by removing the VM volume and remounting it to the new VM. So the cached docker image was there, but the operations the docker image _performed_ did not run again: https://github.com/kubernetes/ingress-nginx/blob/master/rootfs/Dockerfile#L46-L50 TL;DR, had to remove the image so that the image would be rebuilt.
> 
> Even if that solves your problem it's still a problem, but maybe try that? ... Or maybe try setting "imagePullPolicy: Always" and see if that forces the image to be rebuilt?

This Solved my problem, Thanks
